akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowPrefixAndTailSpec.scala

```diff
@@ -192,6 +192,23 @@ class FlowPrefixAndTailSpec extends AkkaSpec {
       upsub.expectCancellation()
     }
 
+    "work even if tail subscriber arrives after substream completion" in {
+      val pub = TestPublisher.manualProbe[Int]()
+      val sub = TestSubscriber.manualProbe[Int]()
+
+      val f = Source(pub).prefixAndTail(1).runWith(Sink.head)
+      val s = pub.expectSubscription()
+      s.sendNext(0)
+
+      val (_, tail) = Await.result(f, 3.seconds)
+
+      val tailPub = tail.runWith(Sink.publisher)
+      s.sendComplete()
```

Comment:
At this point, since prefixAndTail was not seen the input stream completed, it already emitted a substream for the tail instead of Source.empty. We finish the input stream causing the substream to complete.